example/ck_tile/03_gemm/gemm_basic.cpp

```diff
@@ -56,22 +57,29 @@ float gemm_calc(const gemm_basic_args& args, const ck_tile::stream_config& s)
         CShuffleEpilogue,
         ck_tile::CShuffleEpilogue<ck_tile::CShuffleEpilogueProblem<AccDataType,
                                                                    CDataType,
-                                                                   kPadA,
-                                                                   kPadB,
+                                                                   kPadM,
+                                                                   kPadN,
                                                                    kTilePermute,
                                                                    kOutputRank,
                                                                    1,
                                                                    0,
                                                                    TilePartitioner::kM,
                                                                    TilePartitioner::kN>>,
         ck_tile::Default2DEpilogue<
-            ck_tile::Default2DEpilogueProblem<AccDataType, CDataType, kPadA, kPadB>>>;
-
-    using CodegenGemmTraits =
-        ck_tile::TileGemmTraits<kPadA, kPadB, kPadC, ALayout, BLayout, CLayout>;
-    using CodegenPipelineProblem = ck_tile::
-        GemmPipelineProblem<ADataType, BDataType, AccDataType, CodegenGemmShape, CodegenGemmTraits>;
-    using CodegenGemmPolicy = ck_tile::UniversalGemmPipelineAgBgCrPolicy<ALayout, BLayout, CLayout>;
+            ck_tile::Default2DEpilogueProblem<AccDataType, CDataType, kPadM, kPadN>>>;
+
+    using CodegenGemmTraits      = ck_tile::TileGemmTraits<kPadM,
+                                                      kPadN,
+                                                      kPadK,
+                                                      CodegenGemmShape,
+                                                      ADataType,
+                                                      BDataType,
+                                                      AccDataType,
+                                                      ALayout,
+                                                      BLayout,
+                                                      CLayout>;
+    using CodegenPipelineProblem = ck_tile::GemmPipelineProblem<CodegenGemmTraits>;
+    using CodegenGemmPolicy      = ck_tile::UniversalGemmPipelineAgBgCrPolicy;
     using CodegenGemmPipeline =
```

Comment:
Did we tested the modifiled pipeline/policy by giving them more template parameter combinations? e,g,. 16x16x16 warp gemm, block warps=<1, 1, 1>, tile size=32x32x16. It's hard to use `UniversalGemmPipelineAgBgCrPolicy` with the parameters I mentioned. I don't know if you bring the difficulty to other piplelines/policies as well.

Reply:
I have tested with other tile size parameters and warp tile. If we put the unsupported warp or tile size combination, the static_assert should stop that. For the combination selection, I recommend to put that in the CodeEngine to select the supported and best performed combination. We should add that later.